### PR TITLE
Add label to "default quota" multiselect

### LINKED ISSUE #36923: [BITV] 9.1.3.1h/8.2 - The "Default quota" input field in the "Settings" section is not programmatically linked to its visual label. It is recommended to use the text "Default quota", which is currently implemented using a p element, as a label element instead. (1)
- [ ] Fix
- [ ] Backport

4b98244143224a248f230b7ff75c24c8

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-fa5c8d46f9-v8-n2

</details>

### PR DESCRIPTION
**Pull request by JuliaKirschenheuter**
* Resolves: #36923 

- [x] **Waiting for Nc Vue 7.11.3 https://github.com/nextcloud/server/pull/38075#discussion_r1186202461** 

## Summary

Correct label for "default quota" multiselect. No visual changes.

|  Before   | After                                                        |
|-------------|---------------------------------------------------------------|
| Screenshot from 2023-05-04 15-39-41 | Screenshot from 2023-05-09 10-08-18

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by artonge — COMMENTED** [review]
Can we take the opportunity to switch to NcSelect?
https://nextcloud-vue-components.netlify.app/#/Components/NcSelect

**Comment by JuliaKirschenheuter** [comment]
> Can we take the opportunity to switch to NcSelect? https://nextcloud-vue-components.netlify.app/#/Components/NcSelect

I've converted NcMultiselect to NcSelect but have 2 problems: 

1. `:close-on-select="false"` seems not to work. I would like to record this odd behavior but unfortunately can't. Somehow it si not possible.

2. Have a problem with 'unlimited', If i'm choosing it -> 'none' will be shown. After page reloading i get correct 'unlimited'.

@Pytal , @susnux could you please check out this state and help me with correction? Thanks a lot!

**Comment by susnux** [comment]
>     2. Have a problem with 'unlimited', If i'm choosing it -> 'none' will be shown. After page reloading i get correct 'unlimited'.

Because the `setDefaultQuota` sets the label to the id which is *undefined*, one had to `defaultQuota` in this case to `unlimitedQuota`. (I hope it was ok to push this directly? Feel free to squash / revert  :) )

File: apps/settings/src/views/Users.vue [hunk 1]
```diff
@@ -81,17 +81,14 @@
 			<template #footer>
 				<NcAppNavigationSettings>
```
Comment (susnux) [inline apps/settings/src/views/Users.vue:82]:
```suggestion
				<NcAppNavigationSettings :exclude-click-outside-selectors=".vs__dropdown-menu">
```

To fix closing if you click the dropdown menu  you will have to add it to the exclude list of the *click outside* trigger.

File: apps/settings/src/views/Users.vue [hunk 2]
```diff
@@ -81,17 +81,14 @@
...
 							:options="quotaOptions"
-							tag-placeholder="create"
 							:placeholder="t('settings', 'Select default quota')"
-							label="label"
-							track-by="id"
-							:allow-empty="false"
-							:taggable="true"
-							@tag="validateQuota"
-							@input="setDefaultQuota" />
+							:close-on-select="false"
```
Comment (susnux) [inline apps/settings/src/views/Users.vue:89]:
```suggestion
```
Not needed. Probably you want the dropdown (not the settings) to close as soon as you select a quota.

**Review by susnux — CHANGES_REQUESTED** [review]

File: apps/settings/src/views/Users.vue (hunk 2 above)
Comment (JuliaKirschenheuter) [reply apps/settings/src/views/Users.vue:89]: thank you!

File: apps/settings/src/views/Users.vue (hunk 1 above)
Comment (JuliaKirschenheuter) [reply apps/settings/src/views/Users.vue:82]: unfortunately i can't see this prop in current Nc Vue version and can't build server with this prop too ;(

**Comment by JuliaKirschenheuter** [comment]
> Because the `setDefaultQuota` sets the label to the id which is _undefined_, one had to `defaultQuota` in this case to `unlimitedQuota`. (I hope it was ok to push this directly? Feel free to squash / revert :) )

Thank you! Absolutely fine! Works

File: apps/settings/src/views/Users.vue (hunk 1 above)
Comment (susnux) [reply apps/settings/src/views/Users.vue:82]: Its in the docs: https://nextcloud-vue-components.netlify.app/#/Components/App%20containers/NcAppNavigation?id=ncappnavigationsettings

But currently broken (fix is not released yet), to work we have to wait for 7.11.3.

Comment (susnux) [reply apps/settings/src/views/Users.vue:82]: > But currently broken (fix is not released yet), to work we have to wait for 7.11.3.

It is merged and should now work if you rebase the PR :)

File: apps/settings/src/views/Users.vue [hunk 3]
```diff
@@ -79,19 +79,16 @@
...
+							@open="validateQuota"
```
Comment (Pytal) [inline apps/settings/src/views/Users.vue:90]:
```suggestion
							@option:created="validateQuota"
```
Seems to match https://vue-select.org/api/events.html#option-created?

File: apps/settings/src/views/Users.vue [hunk 4]
```diff
@@ -361,6 +358,8 @@ export default {
 		 * @param {string | object} quota Quota in readable format '5 GB' or Object {id: '5 GB', label: '5GB'}
 		 */
 		setDefaultQuota(quota = 'none') {
+			// Make sure correct label is set for unlimited quota
+			if (quota === 'none') quota = this.unlimitedQuota
```
Comment (Pytal) [inline apps/settings/src/views/Users.vue:362]:
```suggestion
			if (quota === 'none') {
				quota = this.unlimitedQuota
			}
```

File: apps/settings/src/views/Users.vue (hunk 3 above)
Comment (Pytal) [reply apps/settings/src/views/Users.vue:90]: And needed the quota arg https://github.com/nextcloud/server/blob/784059fb3800636521d1e8af152bdac7b7f716ae/apps/settings/src/views/Users.vue#L382

Comment (Pytal) [reply apps/settings/src/views/Users.vue:90]: And needed for the quota arg https://github.com/nextcloud/server/blob/784059fb3800636521d1e8af152bdac7b7f716ae/apps/settings/src/views/Users.vue#L382

Comment (JuliaKirschenheuter) [reply apps/settings/src/views/Users.vue]: have i resolved it?

**Review by artonge — APPROVED** [review]
Looks good

**Comment by artonge** [comment]
Have you checked whether the quota is actually changed in the DB?

**Comment by JuliaKirschenheuter** [comment]
> Have you checked whether the quota is actually changed in the DB?

I've checked it by reloading a page and by expecting a same value as set before. Should i double check in DB? Which table should i check for it? Thank you!

File: apps/settings/src/views/Users.vue (hunk 3 above)
Comment (Pytal) [reply apps/settings/src/views/Users.vue]: Don't think it works currently as the `:taggable="true"` is needed as described in https://vue-select.org/api/events.html#option-created

Also you'll have to make sure the quota passed to `validateQuota()` is a string and not an object

**Review by susnux — CHANGES_REQUESTED** [review]
Currently adding custom quotas is broken :/

File: apps/settings/src/views/Users.vue [hunk 5]
```diff
@@ -79,19 +79,16 @@
...
-							tag-placeholder="create"
 							:placeholder="t('settings', 'Select default quota')"
-							label="label"
-							track-by="id"
-							:allow-empty="false"
-							:taggable="true"
-							@tag="validateQuota"
-							@input="setDefaultQuota" />
+							:close-on-select="false"
+							@option:created="validateQuota"
```
[the remark this answers is not in the document]
Comment (JuliaKirschenheuter) [reply apps/settings/src/views/Users.vue:90]: Thank you!

File: apps/settings/src/views/Users.vue [hunk 6]
```diff
@@ -79,19 +79,17 @@
...
 							:options="quotaOptions"
-							tag-placeholder="create"
 							:placeholder="t('settings', 'Select default quota')"
-							label="label"
-							track-by="id"
-							:allow-empty="false"
-							:taggable="true"
-							@tag="validateQuota"
-							@input="setDefaultQuota" />
+							:close-on-select="true"
```
Comment (Pytal) [inline apps/settings/src/views/Users.vue:90]:
```suggestion
```
Can be removed due to being the same as the default https://github.com/nextcloud/nextcloud-vue/blob/6969815a25b44486a3ccd2d86f27e40b6578b40d/src/components/NcSelect/NcSelect.vue#L594-L602

**Review by Pytal — APPROVED** [review]

File: apps/settings/src/views/Users.vue [hunk 7]
```diff
@@ -384,17 +386,21 @@ export default {
 		/**
 		 * Validate quota string to make sure it's a valid human file size
 		 *
-		 * @param {string} quota Quota in readable format '5 GB'
+		 * @param {string | object} quota Quota in readable format '5 GB' or Object {id: '5 GB', label: '5GB'}
 		 * @return {Promise|boolean}
```
Comment (susnux) [inline apps/settings/src/views/Users.vue:390]: Doc type does not match anymore

```suggestion
		 * @return {object} The validated quota object or unlimited quota if input is invalid
```

File: apps/settings/src/views/Users.vue [hunk 8]
```diff
@@ -384,17 +386,21 @@ export default {
...
+			}
 			// only used for new presets sent through @Tag
 			const validQuota = OC.Util.computerFileSize(quota)
 			if (validQuota === null) {
-				return this.setDefaultQuota('none')
+				return this.unlimitedQuota
 			} else {
 				// unify format output
-				return this.setDefaultQuota(OC.Util.humanFileSize(OC.Util.computerFileSize(quota)))
+				quota =  OC.Util.humanFileSize(OC.Util.computerFileSize(quota))
```
Comment (susnux) [inline apps/settings/src/views/Users.vue:402]:
```suggestion
				quota = OC.Util.humanFileSize(OC.Util.computerFileSize(quota))
```

**Review by susnux — APPROVED** [review]
Besides that two minor comments everything is fine :smiley:

**Review by artonge — APPROVED** [review]

**Pull request merged**